File: trunk/C#/PokerProtocol/Commands/Game/PlayerWonPotCommand.cs
using System.Text;
using EricUtility;
using EricUtility.Networking.Commands;

namespace PokerProtocol.Commands.Game
{
    public class PlayerWonPotCommand : AbstractCommand
    {
        protected override string CommandName
        {
            get { return COMMAND_NAME; }
        }
        public static string COMMAND_NAME = "gamePLAYER_WON_POT";

        private readonly int m_PlayerPos;
        private readonly int m_PotID;
        private readonly int m_Shared;
        private readonly int m_PlayerMoney;

        public int PlayerPos
        {
            get { return m_PlayerPos; }
        }
        public int PotID
        {
            get { return m_PotID; }
        }
        public int Shared
        {
            get { return m_Shared; }
        }
        public int PlayerMoney
        {
            get { return m_PlayerMoney; }
        }

        public PlayerWonPotCommand(StringTokenizer argsToken)
        {
            m_PlayerPos = int.Parse(argsToken.NextToken());
            m_PotID = int.Parse(argsToken.NextToken());
            m_Shared = int.Parse(argsToken.NextToken());
            m_PlayerMoney = int.Parse(argsToken.NextToken());
        }

        public PlayerWonPotCommand(int pos, int potID, int shared, int money)
        {
            m_PlayerPos = pos;
            m_PotID = potID;
            m_Shared = shared;
            m_PlayerMoney = money;
        }

        public override void Encode(StringBuilder sb)
        {
            Append(sb, m_PlayerPos);
            Append(sb, m_PotID);
            Append(sb, m_Shared);
            Append(sb, m_PlayerMoney);
        }
    }
}

File: trunk/JAVA/PokerProtocol/src/bluffinmuffin/protocol/commands2/game/PlayerWonPotCommand.java
package bluffinmuffin.protocol.commands2.game;

import java.util.StringTokenizer;

import bluffinmuffin.protocol.commands2.AbstractCommand;

public class PlayerWonPotCommand extends AbstractCommand
{
    @Override
    protected String getCommandName()
    {
        return PlayerWonPotCommand.COMMAND_NAME;
    }
    
    private final int m_playerPos;
    private final int m_potID;
    private final int m_shared;
    private final int m_playerMoney;
    
    public static String COMMAND_NAME = "gamePLAYER_WON_POT";
    
    public PlayerWonPotCommand(StringTokenizer argsToken)
    {
        m_playerPos = Integer.parseInt(argsToken.nextToken());
        m_potID = Integer.parseInt(argsToken.nextToken());
        m_shared = Integer.parseInt(argsToken.nextToken());
        m_playerMoney = Integer.parseInt(argsToken.nextToken());
    }
    
    public PlayerWonPotCommand(int pos, int potID, int shared, int money)
    {
        m_playerPos = pos;
        m_potID = potID;
        m_shared = shared;
        m_playerMoney = money;
    }
    
    @Override
    public void encode(StringBuilder sb)
    {
        append(sb, m_playerPos);
        append(sb, m_potID);
        append(sb, m_shared);
        append(sb, m_playerMoney);
    }
    
    public int getPlayerPos()
    {
        return m_playerPos;
    }
    
    public int getPotID()
    {
        return m_potID;
    }
    
    public int getShared()
    {
        return m_shared;
    }
    
    public int getPlayerMoney()
    {
        return m_playerMoney;
    }
}
